internal/rootcoord/root_coord.go

```diff
@@ -2495,9 +2502,12 @@ func (c *Core) postImportPersistLoop(ctx context.Context, taskID int64, colID in
 	c.wg.Add(1)
 	c.checkSegmentLoadedLoop(ctx, taskID, colID, segIDs)
 	// Check if collection has any indexed fields. If so, start a loop to check segments' index states.
-	c.MetaTable.ddLock.RLock()
-	defer c.MetaTable.ddLock.RUnlock()
-	colMeta := c.MetaTable.collID2Meta[colID]
+	var colMeta etcdpb.CollectionInfo
+	func() {
+		c.MetaTable.ddLock.RLock()
+		defer c.MetaTable.ddLock.RUnlock()
+		colMeta = c.MetaTable.collID2Meta[colID]
+	}()
```

Comment:
Same here, `MetaTable` has an existing method to get collection meta with collectionID

Reply:
Replaced with GetCollectionByID, thanks!